notebooks/403-action-recognition-webcam/403-action-recognition-webcam.ipynb

```diff
@@ -0,0 +1,646 @@
+    "                \n",
+    "                #Decoder inference per set of frames\n",
+    "                # wait for sample duration to work with decoder model\n",
+    "                if len(encoder_output) == sample_duration:\n",
+    "                    decoded_labels, decoded_top_probs = decoder(encoder_output, exec_net_de, input_key_de)\n",
+    "                                        \n",
+    "                #Inference has finished ... Let's to display results\n",
+    "                stop_time = time.time()\n",
+    "                \n",
+    "                # calculate processing time\n",
+    "                processing_times.append(stop_time - start_time)\n",
+    "                # create a text template to show inference results over video\n",
+    "                text_template = '{label},{conf:.2f}%,{Time:.1f}ms,{fps:.1f}FPS'\n",
+    "                # use processing times from last 200 frames\n",
+    "                if len(processing_times) > 200:\n",
+    "                    processing_times.popleft()\n",
+    "                    \n",
+    "                _, f_width = frame.shape[:2]\n",
+    "                # mean processing time [ms]\n",
+    "                processing_time = np.mean(processing_times) * 1000\n",
+    "                fps = 1000 / processing_time\n",
+    "\n",
+    "                #Visualize the results\n",
+    "                for i in range(0, 3):\n",
+    "                    display_text = text_template.format(label=decoded_labels[i], conf=decoded_top_probs[i] * 100, Time=processing_time, fps=fps)\n",
+    "                    text_loc = (TEXT_LEFT_MARGIN, TEXT_VERTICAL_INTERVAL * (i + 1))\n",
+    "                    cv2.putText(frame, display_text, text_loc, FONT_STYLE, FONT_SIZE, FONT_COLOR)\n",
+    "                \n",
+    "                # use this workaround if there is flickering\n",
+    "                if use_popup:\n",
+    "                    cv2.imshow(title, frame)\n",
+    "                    key = cv2.waitKey(1)\n",
+    "                    # escape = 27\n",
+    "                    if key == 27:\n",
+    "                        break\n",
+    "                else:\n",
+    "                    # encode numpy array to jpg\n",
+    "                    _, encoded_img = cv2.imencode(\".jpg\", frame, params=[cv2.IMWRITE_JPEG_QUALITY, 90])\n",
+    "                    # create IPython image\n",
+    "                    i = display.Image(data=encoded_img)\n",
+    "                    # display the image in this notebook\n",
+    "                    display.clear_output(wait=True)\n",
+    "                    display.display(i)\n",
+    "    # ctrl-c\n",
+    "    except KeyboardInterrupt:\n",
+    "        print(\"Interrupted\")\n",
+    "    # any different error\n",
+    "    except RuntimeError as e:\n",
+    "        print(e)\n",
+    "\n",
+    "    # stop capturing\n",
+    "    player.stop()\n",
```

Comment:
Should player.start() and player.stop() be inside try/except/finally?

Reply:
Yes, I could do it, Do you think we should update 401 and 402 notebooks?

Reply:
@adrianboguszewski can you comment on whether this should be in a try/finally?

Reply:
401 is updated to have player.stop() in finally. Paula probably used code from 402, which is not updated, but it works a different way.